python/arcticdb/diagnostics_engine/library_options.py

```diff
@@ -0,0 +1,324 @@
+    info_getter.display_library_options()
+    """
+
+    def __init__(self, lib, is_internal_api, explanations):
+        self.response_obj = lib._lib_cfg.lib_desc.version
+        self.is_internal_api = is_internal_api
+        self.explanations = explanations
+        self.response_non_default_str = (
+            "## Next, see the list of your library options that has been modified with respect to the default"
+            " values:\n\n"
+        )
+        self.response_default_str = (
+            "## Next, see the list of your library options that remains with the default values:\n\n"
+        )
+
+        # Default values defined internally
+        self.c_plus_plus_overwritten_when_zero = {
+            "column_group_size": 127,
+            "segment_row_size": 100000,
+            "max_num_buckets": 150,
+            "max_blob_size": 16777216,
+        }
+
+        self.internal_default_values = {
+            "prune_previous_version": True,
+            "dynamic_strings": True,
+            "use_tombstones": True,
+            "fast_tombstone_all": True,
+            "symbol_list": True,
+        }
+
+        # values that are not a library option, starting from lib._lib_cfg.lib_desc.version
+        self.exclusion_set = {
+            "write_options.sync_disabled.enabled",
+            "write_options.sync_passive.enabled",
+            "write_options.sync_active.enabled",
+            "failure_sim.write_failure_prob",
+            "failure_sim.read_failure_prob",
+            "deleted",
+            "prometheus_config.instance",
+            "prometheus_config.host",
+            "prometheus_config.port",
+            "prometheus_config.job_name",
+            "prometheus_config.prometheus_env",
+            "prometheus_config.prometheus_model",
+            "event_logger_config.logstash_config.event_type",
+            "event_logger_config.logstash_config.host",
+            "event_logger_config.logstash_config.port",
+            "event_logger_config.logstash_config.allow_name_lookup",
+            "event_logger_config.file_logger.file_path",
+            "storage_fallthrough",
+        }
+
+    def traverse_descriptor(self, obj, path=""):
+        """
+        Traverses the fields of the provided protobuf object, compares their values with default ones,
+        and updates response strings accordingly. The function is recursive for nested protobuf message types.
+        """
+        default_response_obj = type(obj)()
+        for field in obj.DESCRIPTOR.fields:
+            value = getattr(obj, field.name)
+            full_name = path + "." + field.name if path else field.name
+            if field.message_type:
+                self.traverse_descriptor(value, full_name)
+            elif full_name not in self.exclusion_set:
+                default_value = (
+                    getattr(default_response_obj, field.name)
+                    if (not self.is_internal_api or field.name not in self.internal_default_values)
```

Comment:
For the external API, we should be displaying everything in terms of the `LibraryOptions` class, not the entire protobuf:
https://docs.arcticdb.io/api/arcticdb/arcticdb.LibraryOptions

Reply:
Ok done. Given that the library options for the external and internal APIs are exposed and modified by the user in different ways, I have completely separated the functions that retrieve the library options for the two APIs. This separation will help avoid unnecessary and messy code. 

Basically, for the external API case, I create a default LibraryOptions object and then traverse all the variables (i.e., options). These are the only ones with which the user can interact. Subsequently, I need to traverse the protobuf object of the library starting from the lib._lib_cfg.lib_desc.version level, in order to find the corresponding value. Please note that not all user options necessarily fall under write_options. For instance, encoding_version is one level up, and there could be future options that we may decide to add.